File: bzk/bzkflow/src/main/java/net/bzk/flow/model/RunLog.java
package net.bzk.flow.model;

import lombok.Data;
import net.bzk.flow.enums.Enums;
import net.bzk.flow.model.var.VarMap;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.IndexDirection;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.util.Date;
import java.util.HashMap;

@SuppressWarnings("serial")
@Data
@Document(collection = "run_log")
public class RunLog implements Serializable {

    @Id
    private String uid;


    private String flowUid;
    @Indexed(name = "flow_uid_index", direction = IndexDirection.DESCENDING)
    private String runFlowUid;
    private String boxUid;
    private String runBoxUid;
    @Indexed(name = "action_uid_index", direction = IndexDirection.DESCENDING)
    private String actionUid;
    private String runActionUid;
    private String refFlowUid;
    @Indexed(name = "ref_run_flow_uid_index",  direction = IndexDirection.ASCENDING)
    private String refRunFlowUid;

    private String msg;

    private VarMap flowVar;

    private VarMap boxVar;
    private String boxName;

    private Enums.RunState state;

    private Enums.LogLv logLv = Enums.LogLv.DEBUG;

    private String exception;

    private String exceptionClazz;

    private HashMap<String, Object> varVals;
    private String actionName;

    @Indexed(name = "create_at_index", direction = IndexDirection.DESCENDING, expireAfterSeconds = 60 * 60 * 20)
    private Date createAt;

}

File: flow-editor/src/app/model/run-log.ts
import { BaseVar } from "../infrastructure/meta";
import { LogLv, RunState, VarLv } from "./pojo/enums";

export class RunLog {

  public id: number;

  public flowUid: string;
  public runFlowUid: string;
  public boxUid: string;
  public runBoxUid: string;
  public actionUid: string;
  public runActionUid: string;
  public refRunFlowUid: string;
  public refFlowUid: string;

  public msg: string;
  public flowVar: BaseVar;
  public boxVar: BaseVar;
  public boxName: string;
  public state = RunState.ActionCall;
  public logLv = LogLv.DEBUG;
  public exception: string;
  public exceptionClazz: string;
  public varVals: Array<VarVal>;
  public actionName: string;

  public updateAt: string;
  public createAt: string;

}

export class VarVal {
  public lv: VarLv;
  public key: string;
  public val: any;
}
